Libraries/PeriphDrivers/Source/SPIMSS/spimss_reva.c

```diff
@@ -231,6 +242,197 @@ int MXC_SPIMSS_RevA_MasterTrans(mxc_spimss_reva_regs_t *spi, spimss_reva_req_t *
+
+                states[i].channelRx = -1;
+                states[i].auto_dma_handlers = false;
+                states[i].req = NULL;
+            }
+
+            if (NULL != temp_req->callback) {
+                temp_req->callback(temp_req, E_NO_ERROR);
+            }
+            break;
+        }
+    }
+}
+
+static mxc_dma_config_t dma_config_tx;
+static mxc_dma_config_t dma_config_rx;
+static mxc_dma_adv_config_t dma_adv_config_tx;
+static mxc_dma_adv_config_t dma_adv_config_rx;
+static mxc_dma_srcdst_t srcdst_config_tx;
+static mxc_dma_srcdst_t srcdst_config_rx;
+/* ************************************************************************** */
+int MXC_SPIMSS_RevA_MasterTransDMA(mxc_spimss_reva_regs_t *spi, spimss_reva_req_t *req)
+{
+    // Calculating the transaction size in byte.
+    int sent_byte_len = ((req->len * (req->bits / 8)) - req->tx_num);
+    int tx_channel_id = -1;
+    int rx_channel_id = -1;
+    int spi_num = MXC_SPIMSS_GET_IDX((mxc_spimss_regs_t *)spi);
+
+    //Reset static variables.
+    // tx_channel_reg = NULL;
+    // rx_channel_reg = NULL;
+    memset(&dma_config_tx, 0, sizeof(dma_config_tx));
+    memset(&dma_config_rx, 0, sizeof(dma_config_rx));
+    memset(&srcdst_config_tx, 0, sizeof(srcdst_config_tx));
+    memset(&srcdst_config_rx, 0, sizeof(srcdst_config_rx));
+    memset(&dma_adv_config_tx, 0, sizeof(dma_adv_config_tx));
+    memset(&dma_adv_config_rx, 0, sizeof(dma_adv_config_rx));
+
+    // We use SPIMSS master trans function to send data from
+    if (E_NO_ERROR != MXC_SPIMSS_RevA_TransSetup(spi, req, 1)) {
+        return -1;
+    }
+
+    //Setting dmas fifo levels for tx and rx in spimss.
+    spi->dma &= ~(7 << MXC_F_SPIMSS_DMA_RX_FIFO_LVL_POS);
+    spi->dma &= ~(7 << MXC_F_SPIMSS_DMA_TX_FIFO_LVL_POS);
+
+    //Initialization of DMA
+    MXC_DMA_Init();
+
+    if ((true == states[spi_num].auto_dma_handlers) && (0 > states[spi_num].channelTx)) {
+        // Getting the available tx channel id.
+        tx_channel_id = MXC_DMA_AcquireChannel();
+        if (tx_channel_id < 0) {
+            return -1;
+        }
+        MXC_SPIMSS_RevA_SetTXDMAChannel(spi, tx_channel_id);
+
+        //Enable NVIC interrupt of the dma channel which will be used as TX channel.
+        //NVIC_EnableIRQ(MXC_DMA_CH_GET_IRQ(tx_channel_id));
+        MXC_NVIC_SetVector(MXC_DMA_CH_GET_IRQ(tx_channel_id), MXC_DMA_Handler);
+    } else {
+        tx_channel_id = MXC_SPIMSS_RevA_GetTXDMAChannel(spi);
+        if (tx_channel_id < 0) {
+            return E_BAD_STATE;
+        }
+    }
+
+    if ((true == states[spi_num].auto_dma_handlers) && (0 > states[spi_num].channelRx)) {
+        rx_channel_id = MXC_DMA_AcquireChannel();
+        if (rx_channel_id < 0) {
+            return -1;
+        }
+        MXC_SPIMSS_RevA_SetRXDMAChannel(spi, rx_channel_id);
+
+        //Enable NVIC interrupt of the dma channel which will be used as RX channel.
+        //NVIC_EnableIRQ(MXC_DMA_CH_GET_IRQ(rx_channel_id));
```

Comment:
We need to enable the IRQ here, so I think this should be uncommented

Reply:
MXC_NVIC_SetVector calls NVIC_EnableIRQ inside. So, I didn't want to call NVIC_EnableIRQ again.